### PR TITLE
Tracing propagation

### PR DESCRIPTION
**Pull request by tomas-langer**
Documentation on MP tracing and tracing propagation.

### CODE REVIEW

File: docs/src/main/docs/tracing/01_tracing.adoc [hunk 1]
```diff
@@ -57,3 +57,72 @@ ServerConfiguration.builder()
 ----
 <1> The name of the application (service) to associate with the tracing events
 <2> The endpoint for tracing events, specific to the tracer used, usually loaded from Config
+
+=== Configuring Tracing with Helidon MP
+There is an additional dependency that adds Tracing support to MP server and Jersey client:
+
+[source,xml,subs="verbatim,attributes"]
```
Comment (romain-grecourt) [inline docs/src/main/docs/tracing/01_tracing.adoc]: I know this is used everywhere, `subs="verbatim,attributes` is not needed here (and in most cases).
Will need to do a pass to clean this up everywhere in the docs, but let's not add more ;)

File: docs/src/main/docs/tracing/01_tracing.adoc [hunk 2]
```diff
@@ -57,3 +57,72 @@ ServerConfiguration.builder()
 ----
 <1> The name of the application (service) to associate with the tracing events
 <2> The endpoint for tracing events, specific to the tracer used, usually loaded from Config
+
+=== Configuring Tracing with Helidon MP
```
Comment (romain-grecourt) [inline docs/src/main/docs/tracing/01_tracing.adoc]: Shouldn't this be instead a separate document under `microprofile/` ?

File: docs/src/main/docs/tracing/01_tracing.adoc [hunk 3]
```diff
@@ -57,3 +57,72 @@ ServerConfiguration.builder()
...
+There is an additional dependency that adds Tracing support to MP server and Jersey client:
```
Comment (romain-grecourt) [inline docs/src/main/docs/tracing/01_tracing.adoc]: The first sentence should describes the "tracing support to MP server and Jersey client".
Then you can just add a simple sentence in a separate paragraph like "Declare the following dependency in your project".

File: docs/src/main/docs/tracing/01_tracing.adoc [hunk 4]
```diff
@@ -57,3 +57,72 @@ ServerConfiguration.builder()
...
+
+[source,xml,subs="verbatim,attributes"]
+.MP Tracing
```
Comment (romain-grecourt) [inline docs/src/main/docs/tracing/01_tracing.adoc]: I'd remove this title

File: docs/src/main/docs/tracing/01_tracing.adoc [hunk 5]
```diff
@@ -57,3 +57,72 @@ ServerConfiguration.builder()
...
+----
+<dependency>
+    <groupId>io.helidon.microprofile.tracing</groupId>
+    <artifactId>helidon-microprofile-tracing</artifactId>
+</dependency>
+----
+
+The minimal required configuration is `tracing.service` that contains
```
Comment (romain-grecourt) [inline docs/src/main/docs/tracing/01_tracing.adoc]: As I understand this part explains how to configure tracing for Helidon MP. I'd rather suggest a specific section about configuration.

Show all the configuration options with a description, and specify which ones are optional or required ; A table is a good fit unless there is a large number of configuration options, in this case I'd show the most common one and refer to the full list in the javadocs with a link.

File: docs/src/main/docs/tracing/01_tracing.adoc [hunk 6]
```diff
@@ -57,3 +57,72 @@ ServerConfiguration.builder()
...
+- When you are not in a Jersey Server scope (e.g. when using Jersey client from Helidon SE)
+
+In such cases, you must provide the SpanContext by hand:
+[source,java]
+.Tracing propagation with Jersey client (on a different thread)
+----
+import static io.helidon.tracing.jersey.client.ClientTracingFilter.CURRENT_SPAN_CONTEXT_PROPERTY_NAME;
+import static io.helidon.tracing.jersey.client.ClientTracingFilter.TRACER_PROPERTY_NAME;
+
+...
```
Comment (romain-grecourt) [inline docs/src/main/docs/tracing/01_tracing.adoc]: Please use `//...`, it's better for syntax highlighting.

File: docs/src/main/docs/tracing/01_tracing.adoc (hunk 3 above)
Comment (tomas-langer) [reply docs/src/main/docs/tracing/01_tracing.adoc]: Fixed

File: docs/src/main/docs/tracing/01_tracing.adoc (hunk 4 above)
Comment (tomas-langer) [reply docs/src/main/docs/tracing/01_tracing.adoc]: Fixed

File: docs/src/main/docs/tracing/01_tracing.adoc (hunk 3 above)
Comment (tomas-langer) [reply docs/src/main/docs/tracing/01_tracing.adoc]: Reworded

File: docs/src/main/docs/tracing/01_tracing.adoc (hunk 2 above)
Comment (tomas-langer) [reply docs/src/main/docs/tracing/01_tracing.adoc]: yes, fixed

File: docs/src/main/docs/tracing/01_tracing.adoc (hunk 1 above)
Comment (tomas-langer) [reply docs/src/main/docs/tracing/01_tracing.adoc]: removed

File: docs/src/main/docs/tracing/01_tracing.adoc (hunk 5 above)
Comment (tomas-langer) [reply docs/src/main/docs/tracing/01_tracing.adoc]: Added section with configuration options.

File: docs/src/main/docs/tracing/01_tracing.adoc (hunk 6 above)
Comment (tomas-langer) [reply docs/src/main/docs/tracing/01_tracing.adoc]: Fixed

File: docs/src/main/docs/microprofile/07_tracing.adoc [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+@Context
+io.helidon.webserver.ServerRequest serverRequest;
+
+//...
+
+Span span = GlobalTracer.get()
+        .buildSpan("my-operation")
+        .asChildOf(serverRequest.spanContext())
+        .start();
+----
```
Comment (spericas) [inline docs/src/main/docs/microprofile/07_tracing.adoc]: With the recent activation change, the new span should become part of the server request span, right? Perhaps we can add a note to that. I haven't actually tried this with MP, though.

File: docs/src/main/docs/microprofile/07_tracing.adoc [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+@Context
+io.helidon.webserver.ServerRequest serverRequest;
+
+//...
+
+SpanContext spanContext = serverRequest.spanContext();
+// optional, you could also use GlobalTracer.get() if it is configured
+Tracer tracer = req.webServer().configuration().tracer();
+----
```
Comment (spericas) [inline docs/src/main/docs/microprofile/07_tracing.adoc]: Should we add a shorthand method in `ServerRequest` to get the tracer?

File: docs/src/main/docs/tracing/01_tracing.adoc [hunk 9]
```diff
@@ -57,3 +55,64 @@ ServerConfiguration.builder()
...
+[source,java]
+----
+// or you can register the tracer as a global tracer
+// and use GlobalTracer.get()
+Tracer tracer = request.webServer().configuration().tracer();
+Span span = tracer
+        .buildSpan("my-operation")
+        .asChildOf(serverRequest.spanContext())
+                         .start();
+----
```
Comment (spericas) [inline docs/src/main/docs/tracing/01_tracing.adoc]: Ditto comments in previous file.

File: docs/src/main/docs/microprofile/07_tracing.adoc (hunk 8 above)
Comment (tomas-langer) [reply docs/src/main/docs/microprofile/07_tracing.adoc]: Sounds like  a good idea, as we do not automatically update the `GlobalTracer`, we may want to add the shortcut.

**Comment by tomas-langer** [comment]
I will merge this and we should revisit this documentation once we have MP Tracing implemented.

**Pull request merged**